File: uri_1094.py
qnt = int(input())

cobaias = []

coelhos, ratos, sapos, total = 0, 0, 0, 0

for i in range(qnt):
    linha = input().split()
    cobaias.append((int(linha[0]), linha[1]))

for i in cobaias:
    total += i[0]

    if i[1] == 'C':
        coelhos += i[0]
    elif i[1] == 'R':
        ratos += i[0]
    else:
        sapos += i[0]

print(f'Total: {total:d} cobaias')
print(f'Total de coelhos: {coelhos:d}')
print(f'Total de ratos: {ratos:d}')
print(f'Total de sapos: {sapos:d}')
print(f'Percentual de coelhos: {coelhos*100/total:.2f} %')
print(f'Percentual de ratos: {ratos*100/total:.2f} %')
print(f'Percentual de sapos: {sapos*100/total:.2f} %')

File: Uri1094.java
package Uri;

public class Uri1094 {

    public static void main(String[] args) {
        java.util.Scanner l = new java.util.Scanner(System.in);

        int qnt;
        int totalDeCobaias = 0, coelhos = 0, sapos = 0, ratos = 0;
        String type;

        int v = l.nextInt();

        for (int i = 1; i <= v; i++) {
            qnt = l.nextInt();
            type = l.nextLine();
            
            totalDeCobaias += qnt;

            switch (type) {
                case " C":
                    coelhos += qnt;
                    break;
                case " R":
                    ratos += qnt;
                    break;
                case " S":
                    sapos += qnt;
                    break;
            }
        }
        System.out.printf("Total: %d cobaias\n", totalDeCobaias);
        System.out.printf("Total de coelhos: %d\n", coelhos);
        System.out.printf("Total de ratos: %d\n", ratos);
        System.out.printf("Total de sapos: %d\n", sapos);
        System.out.printf("Percentual de coelhos: %.2f ", (float) (100 * coelhos) /totalDeCobaias);
        System.out.println("%");
        System.out.printf("Percentual de ratos: %.2f ", (float) (100 * ratos) / totalDeCobaias);
        System.out.println("%");
        System.out.printf("Percentual de Sapos: %.2f ", (float) (100 * sapos) / totalDeCobaias);
        System.out.println("%");
    }


}
